### DISCUSSION TITLE
Ubuntu server 22.04 and Google Coral USB

### DISCUSSION THREAD

**Discussion by Cold-Lemonade** [blakeblackshear/frigate#5915] [General]
Please forgive the naivety of this question--I am an a beginner when it comes to Linux. I understand that Google Coral UBS supports an older version of Python (I believe ver. 3.9) than Ubuntu Server 22.04. To get the Google Coral to work with Frigate on Ubuntu Server 22.04, do I need to downgrade the version of Python it's running? There is no mention of this issue in the Frigate documentation so I am posting this question for clarification. Thank you.

**Comment by NickM-27** [comment]
Frigate runs in docker, meaning everything it needs to run is inside the container and isolated from your system.

**Reply by Cold-Lemonade** [reply]
Thank you @NickM-27 So all I need to do is plug in the coral USB and make sure that my frigate.yaml and config.yml files identify the Google Coral USB and I should be all set?

Separately, I want to express my gratitude to you, @blakeblackshear, and everyone else who has contributed to making Frigate. It is very impressive.

I have been running version 0.11.1. Everything was great since October and then my system froze a couple of days ago. I am not sure what happened but suspect it has something to do with how I had combined an external USB hard drive with and internal one as a single logical drive. When I went to reboot it, I got a kernel panic error message and it seems my hard drives are out of sync with each other. So I am in the process of reinstalling everything. I guess it is a good opportunity to move over to 0.12 RC2. I have been following the progress and it sounds light a major leap forward over 0.11.1. I'm very excited about it. Many thanks to everyone.

**Reply by NickM-27** [reply]
Yes, for the USB coral that is all that is needed. 0.12 should become the stable release quite soon as well

**Reply by thesebastianf** [reply]
is this also true for a M2 A+E Coral?
Can I pass the device to the container and its fine?

**Reply by NickM-27** [reply]
No the driver needs to be installed on the host to have a device to pass through

**Reply by thesebastianf** [reply]
mhh I could pass the RAW device from Proxmox to Debian and thought I just install the drivers in Debian.
But seems like there is currently a issue that some packages from the official coral documentation cant be installed on a recent linux kernel. 
Not sure if it would work at all ... all tutorials I find also make changes to the Proxmox Host OS ...  which I thought I dont need when I passthrough the raw device ( which works).   when I need to tweak proxmox AND debian ... I probably go back to the USB solution :-/

**Comment by Cold-Lemonade** [comment]
Sorry for a follow up question--please pardon my newbie ignorance. To get the USB corals to work with Frigate, do I need to install the Edge TPU runtime (i.e., https://coral.ai/docs/accelerator/get-started/#1-install-the-edge-tpu-runtime)?

**Reply by NickM-27** [reply]
no, frigate has that in the container as well

**Comment by Adam-Al-Rahman** [comment]
Installing Coral USB Accelerator on Ubuntu 22.04 and Python 3.10, follow the below blog instructions
[Coral USB On Ubuntu 22 with python 3.10](https://igor.technology/installing-coral-usb-accelearator-python-3-10-ubuntu-22/)

**Reply by r-zig** [reply]
I'm run it on microk8s on Ubuntu 22.04.2 LTS with Intel_R_ Client Systems NUC12WSHi5.
Shell into the frigate pod does not have lsusb, but under other running pod it output's this usb, even differently than directly on my machine with k8s

Any suggestions?
Thanks
